### PR TITLE
Add domain validation when creating room with list of invitees

### PR DESCRIPTION
**Pull request by werner291**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

This results in a 400 bad request status instead of 500 internal server error in the case described in #4088 (trailing whitespace).

Also adds a regression test for this issue.

Question: An original (abandoned) attempt to fix this issue (see https://github.com/matrix-org/synapse/pull/4351/files) also does some trial encoding/decoding with IDNA. Should we try to do that as well? (See the TODO entry)

### CODE REVIEW

File: synapse/handlers/room.py [hunk 1]
```diff
@@ -554,7 +555,9 @@ def create_room(self, requester, config, ratelimit=True, creator_join_profile=No
         invite_list = config.get("invite", [])
         for i in invite_list:
             try:
-                UserID.from_string(i)
+                uid = UserID.from_string(i)
+                # TODO Maybe attempt idna encoding/decoding as well?
```
Comment (anoadragon453) [inline synapse/handlers/room.py]: @richvdh Do you have any thoughts on this?

File: tests/rest/client/v1/test_rooms.py [hunk 2]
```diff
@@ -485,6 +485,14 @@ def test_post_room_invalid_content(self):
         self.assertEquals(400, channel.code)
 
 
+    def test_post_room_invitees_invalid_mxid(self):
+        # POST with invalid invitee, see https://github.com/matrix-org/synapse/issues/4088
+        # Note the trailing slash in the MXID here!
```
Comment (anoadragon453) [inline tests/rest/client/v1/test_rooms.py]:
```suggestion
        # Note the trailing space in the MXID here!
```

File: synapse/handlers/room.py (hunk 1 above)
Comment (richvdh) [reply synapse/handlers/room.py]: I do have thoughts on it. https://matrix.org/docs/spec/appendices#server-name is clear that server_names (and hence mxids) may contain only ascii: it's up to the client to do idna-encoding/decoding on them if it so desires. 

Please can we get rid of the todo!

Comment (richvdh) [reply synapse/handlers/room.py]:
```suggestion
```

**Pull request merged**